### DISCUSSION TITLE
migration from 2.6.5 --> 2.7.10

### DISCUSSION THREAD

**Discussion by dstoeckm** [netbox-community/netbox-docker#402] [Getting Help] [answered]
is there a more detailed description regarding the migration? I used docker-compose.override.yml to use image: postgres:10.4-alpine. So that worked, but my logs show "nginx_1          | 2021/01/25 17:03:08 [error] 29#29: *2 connect() failed (113: Host is unreachable) while connecting to upstream"

**Comment by cimnine** [comment, marked as answer]
Have you had a look at [our troubleshooting guide](https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting) already? For example, did you check if all the containers started with `docker-compose ps`? What does the log of Netbox say, i.e. `docker-compose logs netbox` (or `docker-compose logs -f netbox` if you want to look at it live as it happens).

**Reply by dstoeckm** [reply]
yes I did. 
I dumped from version2.6.5 by
"docker-compose exec -T postgres sh -c 'pg_dump -cU $POSTGRES_USER $POSTGRES_DB' | gzip > netbox_db_dump.sql.gz"
Installed the latest 2.10 with docker-compose.override.yml for postgres:10.4-alpine all is running fine. when I uploded the dump by "gunzip -c netbox_db_dump.sql.gz | docker-compose exec -T postgres sh -c 'psql -U $POSTGRES_USER $POSTGRES_DB'" I'll get a UI error like
![image](https://user-images.githubusercontent.com/9450000/105762973-6ff33200-5f55-11eb-9755-eeb5b5891aad.png)
[logs_after.txt](https://github.com/netbox-community/netbox-docker/files/5869526/logs_after.txt)
[logs_before.txt](https://github.com/netbox-community/netbox-docker/files/5869527/logs_before.txt)
the logs are before the dump upload and after.

**Reply by cimnine** [reply]
If you update, it is advised that you follow the minor versions, i.e. 2.6 -> 2.7 -> 2.8 -> 2.9 -> 2.10. You should use the corresponding version of Netbox Docker, see our release notes. Also carefully follow the release notes of Netbox itself.

Also – since you have not written this explicitly – make sure only the PostgreSQL container is running when you restore from a DB backup: `docker-compose down && docker-compose up postgres`. Start the other containers only after the restore is complete. Netbox Docker will then do the database migration.

**Reply by cimnine** [reply]
Also, from your logs, it is clear that the database migration did not happen. I believe this is because you have started Netbox 2.10, then restored the data from 2.6.

Because the database was created for 2.10, and after you restored the data from 2.6, Netbox did not apply the database migrations, because it believed that the database (and the data respectively) is already in 2.10 format.

**Reply by dstoeckm** [reply]
ok, to simplify I cloned my docker host (avoiding DB restore). I did "git checkout 0.20.0" "export VERSION=v2.6.7" and did a docker-compos up -d. so far so good all inventory s in my UI, showing 2.6.7 bottom left. Now I do docker-compose down and  "git checkout 0.21.0" "export VERSION=v2.7.0" docker-compose up -d. ==> docker-compose ps: netbox-docker_postgres_1        docker-entrypoint.sh postgres    Exit 1
docker-compose logs -f -t postgres
Attaching to netbox-docker_postgres_1
postgres_1       | 2021-01-29T18:01:27.872259000Z PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres_1       | 2021-01-29T18:01:27.913152000Z 2021-01-29 18:01:27.911 UTC [1] FATAL:  database files are incompatible with server
postgres_1       | 2021-01-29T18:01:27.913558000Z 2021-01-29 18:01:27.911 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 10, which is not compatible with this version 11.10.
netbox-docker_postgres_1 exited with code 1


what i'm missing?

**Reply by cimnine** [reply]
It's right in front of you:

> postgres_1 | 2021-01-29T18:01:27.913558000Z 2021-01-29 18:01:27.911 UTC [1] DETAIL: The data directory **was initialized by PostgreSQL version 10, which is not compatible with this version 11.10**.

You must either continue to use PostgreSQL 10 (override the image in `docker-compose.override.yml`) or correctly update your database. (IMO the easiest way to do it is to backup from old postgres version and restore in new version).

**Reply by dstoeckm** [reply]
exactly the point. Why does the DB migration not kick in?

**Reply by cimnine** [reply]
Because all that Netbox does is upgrade it's _schema_ (i.e. it's table, etc.). But if you upgrade PostgreSQL, you need to upgrade _the database itself_. Afaik PostgreSQL will re-organize it's files on the disk and their internal structure.

Or in other words: It's not something that can be done via SQL, so Netbox (and thous Netbox Docker) can't do it for you. It's the job of the DB admin (you?) to perform the upgrade steps. Please see to the [official PostgreSQL _Upgrading_ documentation](https://www.postgresql.org/docs/current/upgrading.html) to learn more.

Also, be sure to read our Release Notes when upgrading. They clearly state that you need to update PostgreSQL yourself, e.g. [the release notes for 0.26.0](https://github.com/netbox-community/netbox-docker/releases/tag/0.26.0) or [those for 0.20.0](https://github.com/netbox-community/netbox-docker/releases/tag/0.20.0).
